### PR TITLE
[fix] pylint errors on modules packaging language

### PR DESCRIPTION
**Pull request by 4383**
##### SUMMARY
fix pylint errors  on modules packaging language files

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/modules/packaging/language/*.py

##### ANSIBLE VERSION

```
ansible 2.5.0 (pep8/language 01b2454984) last updated 2017/09/22 14:06:02 (GMT +200)
  config file = None
  configured module search path = ['/home/herve/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/herve/dev/ansible/lib/ansible
  executable location = /home/herve/dev/ansible/bin/ansible
  python version = 3.5.2 (default, Nov 17 2016, 17:05:23) [GCC 5.4.0 20160609]

```

##### ADDITIONAL INFORMATION

before changes
```
Sanity check using pylint
ERROR: Found 54 pylint issue(s) which need to be resolved:
ERROR: lib/ansible/modules/packaging/language/bower.py:220:11: len-as-condition Do not use `len(SEQUENCE)` as condition value
ERROR: lib/ansible/modules/packaging/language/bower.py:225:11: len-as-condition Do not use `len(SEQUENCE)` as condition value
ERROR: lib/ansible/modules/packaging/language/bower.py:225:27: len-as-condition Do not use `len(SEQUENCE)` as condition value
ERROR: lib/ansible/modules/packaging/language/bundler.py:138:4: no-else-return Unnecessary "else" after "return"
ERROR: lib/ansible/modules/packaging/language/cpanm.py:142:4: no-else-return Unnecessary "else" after "return"
ERROR: lib/ansible/modules/packaging/language/cpanm.py:177:4: no-else-return Unnecessary "else" after "return"
ERROR: lib/ansible/modules/packaging/language/gem.py:118:4: no-else-return Unnecessary "else" after "return"
ERROR: lib/ansible/modules/packaging/language/maven_artifact.py:192:7: len-as-condition Do not use `len(SEQUENCE)` as condition value
ERROR: lib/ansible/modules/packaging/language/maven_artifact.py:222:8: no-else-return Unnecessary "else" after "return"
ERROR: lib/ansible/modules/packaging/language/maven_artifact.py:228:8: no-else-return Unnecessary "else" after "return"
ERROR: lib/ansible/modules/packaging/language/maven_artifact.py:241:8: no-else-return Unnecessary "else" after "return"
ERROR: lib/ansible/modules/packaging/language/maven_artifact.py:296:88: len-as-condition Do not use `len(SEQUENCE)` as condition value
ERROR: lib/ansible/modules/packaging/language/maven_artifact.py:297:86: len-as-condition Do not use `len(SEQUENCE)` as condition value
ERROR: lib/ansible/modules/packaging/language/maven_artifact.py:345:8: no-else-return Unnecessary "else" after "return"
ERROR: lib/ansible/modules/packaging/language/maven_artifact.py:347:12: no-else-return Unnecessary "else" after "return"
ERROR: lib/ansible/modules/packaging/language/maven_artifact.py:386:8: no-else-return Unnecessary "else" after "return"
ERROR: lib/ansible/modules/packaging/language/npm.py:267:11: len-as-condition Do not use `len(SEQUENCE)` as condition value
ERROR: lib/ansible/modules/packaging/language/npm.py:273:11: len-as-condition Do not use `len(SEQUENCE)` as condition value
ERROR: lib/ansible/modules/packaging/language/npm.py:276:11: len-as-condition Do not use `len(SEQUENCE)` as condition value
ERROR: lib/ansible/modules/packaging/language/pear.py:87:4: no-else-return Unnecessary "else" after "return"
```

```
Sanity check using no-wildcard-import
Sanity check using pep8
Sanity check using pylint
Sanity check using pylint-ansible-test
```

### CODE REVIEW

**Review by mwarkentin — APPROVED** [review]
Changes LGTM.

File: lib/ansible/modules/packaging/language/bundler.py
Comment (abadger) [inline lib/ansible/modules/packaging/language/bundler.py]: This change isn't good as it makes the module fail if the bundle command isn't found.  need to keep the original if-else structure but can change it to return result.  I also only like to pre-load one answer if the pre-loading is a simple assignment as those are both cheap and do not have side-effects.  In this case, the function being called has a side-effect; in other cases a function being called might have a performance impact. Something like this should satisfy pylint but not suffer these problems:
``` python
def get_bundler_executable(module):
    if module.params.get('executable'):
        result = module.params.get('executable').split(' ')
    else:
        result = [ module.get_bin_path('bundle', True) ]
    return result
```

Comment (4383) [reply lib/ansible/modules/packaging/language/bundler.py]: @abadger fix done!

File: lib/ansible/modules/packaging/language/pear.py
Comment (abadger) [inline lib/ansible/modules/packaging/language/pear.py]: Same note here as for bundler

File: lib/ansible/modules/packaging/language/cpanm.py
Comment (abadger) [inline lib/ansible/modules/packaging/language/cpanm.py]: Same note here as for bundler

File: lib/ansible/modules/packaging/language/gem.py
Comment (abadger) [inline lib/ansible/modules/packaging/language/gem.py]: same note here as for bundler

**Comment by abadger** [comment]
@4383 All of my notes are similar to the bundler one (involving module.get_bin_path).  Once those are all fixed, this can be merged.

**Comment by 4383** [comment]
@abadger fix done!

**Comment by abadger** [comment]
Excellent!  I'll merge to devel once shippable passes.

**Pull request merged**
